### PR TITLE
zephyrCommon: Implement analogRead()/analogReference() 

### PR DESCRIPTION
**Pull request by soburi**
The 'io-channels' node under the 'zephyr,user' defines
available ADC channels.

The 'io-channel-pins' defines the mapping of ADC channels
to pin numbers.  API looks up the 'io-channel-pins' when
querying the ADC channel by digital pin number.
So it should be in the same order as 'io-channels' node.

For original Arduino API compatibility, ADC resolution
should configure as 10-bit.

ADC that is on MCU supported by Zephyr exists
only 16-bit resolution currently.
So it supports resolutions up to 16-bit.

Signed-off-by: TOKITA Hiroshi <tokita.hiroshi@fujitsu.com>

### CODE REVIEW

**Review by DhruvaG2000 — COMMENTED** [review]
This PR adds a big chunk of code to overlay files, hence for adding further variants it maybe useful to have documentation for what all is needed to get analog working in these overlays.
Do you think you can also add documentation for the same?

File: samples/analog_input/prj.conf [hunk 1]
```diff
@@ -0,0 +1,4 @@
+CONFIG_GPIO=y
+CONFIG_ADC=y
```
Comment (DhruvaG2000) [inline samples/analog_input/prj.conf:2]: I think all this can be shifted into the project's global KConfig

**Review by DhruvaG2000 — CHANGES_REQUESTED** [review]

File: samples/analog_input/README.rst [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+For example, to build this sample for :ref:`arduino_nano_33_ble`:
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/basic/arduino-threads
+   :board: arduino_nano_33_ble
+   :goals: build flash
+   :compact:
+
+Change ``arduino_nano_33_ble`` appropriately for other supported boards.
```
Comment (DhruvaG2000) [inline samples/analog_input/README.rst:47]: This README needs updating.

**Review by DhruvaG2000 — CHANGES_REQUESTED** [review]

File: samples/analog_input/prj.conf (hunk 1 above)
Comment (soburi) [reply samples/analog_input/prj.conf:1]: Fixed it.

File: samples/analog_input/README.rst (hunk 2 above)
Comment (soburi) [reply samples/analog_input/README.rst]: I forgot it.😱

Update README did.

**Comment by soburi** [comment]
@DhruvaG2000 
Rebase to newest main branch done.
Could you please review again?

**Comment by soburi** [comment]
> Code looks good, but let me test this patch on real hardware before approving and merging.

Okay, No need to hurry

**Review by DhruvaG2000 — APPROVED** [review]
Built and tested on my arduino nano 33 ble sense, and it works OK.

**Pull request merged**